Python impl

grisumbras opened the pull request:
Python implementation. Collects data, then outputs using a Jinja template.

vinniefalco commented:
You have write access, so I would suggest merging it. But if you could squash it all down to one commit it would make more sense. Although if you don't squash it, that is also OK. As long as this doesn't break anything.

grisumbras commented:
Current differences:

Improvements:
* noexcept support;
* explicit support;
* ref qualifier support;
* hyperlinked types in declarations;
* old implementation had some extra whitespace in code blocks;
* underlying types of enumerations are shown;
* fixed array references in function parameters;
* fixed grouping bug (it resulted in some functions documented several times);
* scoped enums are declared as 'enum class';
* defaulted functions are marked as such.

Minor differences:
* type declarations end with a semicolon similar to function declarations;
* instead of NBSP character a UTF16 escape representing NBSP is output;
* order of operators changes in the summary (my implementation sorts alphabetically, I don't understand the old logic);
* overload index follows overload order in the summary (affects links);
* different style of displaying bases.

Major differences:
* aliases aren't unwrapped;
* doesn't merge member functions and friends (affects links).
